modules/core/src/lib/effect-manager.ts

```diff
@@ -16,9 +17,24 @@ export default class EffectManager {
     this._setEffects([]);
   }
 
-  addDefaultEffect(effect: Effect) {
-    if (!this._defaultEffects.find(e => e.constructor === effect.constructor)) {
-      this._defaultEffects.push(effect);
+  // TODO - better Effect API, may be cleaner to add `order` to `Effect` class
+  /**
+   * Register a new default effect, i.e. an effect present regardless of user supplied props.effects
+   */
+  addDefaultEffect(
+    /** Effect instance. The same effect can only be added once. */
+    effect: Effect,
+    /** Used to sort effects. In case of conflict, effects will be used in the order of registration. */
+    order: number = Infinity
+  ) {
+    const defaultEffects = this._defaultEffects;
+    if (!defaultEffects.find(e => e.effect.constructor === effect.constructor)) {
+      const index = defaultEffects.findIndex(e => e.order > order);
+      if (index < 0) {
+        defaultEffects.push({effect, order});
+      } else {
+        defaultEffects.splice(index, 0, {effect, order});
+      }
```

Comment:
Maybe just push and sort, keep this kind of non-perf critical code as simple as possible.
```suggestion
        defaultEffects.push({effect, order});
        defaultEffects.sort(({order: order1}, {order: order2}) => order1 > order2);
```

Reply:
This implementation preserves the order of registration in the case where `order` is not supplied or collides. If I remember right, `Array.sort` is not stable when it comes to equal elements.